packages/vuetify/src/composables/goto.ts

```diff
@@ -0,0 +1,164 @@
+    totalOffset += horizontal ? el.offsetLeft : el.offsetTop
+    el = el.offsetParent as HTMLElement
+  }
+
+  return totalOffset
+}
+
+export function createGoTo (options: Partial<GoToOptions> | undefined, locale: LocaleInstance & RtlInstance) {
+  return {
+    rtl: locale.isRtl,
+    options: mergeDeep(genDefaults(), options),
+  }
+}
+
+async function scrollTo (
+  _target: ComponentPublicInstance | HTMLElement | number | string,
+  _options: Partial<GoToOptions>,
+  horizontal?: boolean,
+  goTo?: GoToInstance,
+) {
+  const options = mergeDeep(goTo?.options, _options)
+  const rtl = goTo?.rtl.value
+  const target = (typeof _target === 'number' ? _target : getTarget(_target)) ?? 0
+  const container = options.container === 'parent' && target instanceof HTMLElement
+    ? target.parentElement!
+    : getContainer(options.container)
+  const ease = typeof options.easing === 'function' ? options.easing : options.patterns[options.easing]
+
+  if (!ease) throw new TypeError(`Easing function "${options.easing}" not found.`)
+
+  let targetLocation: number
+  if (typeof target === 'number') {
+    targetLocation = getOffset(target, horizontal, rtl)
+  } else {
+    targetLocation = getOffset(target, horizontal, rtl) - getOffset(container, horizontal, rtl)
+
+    if (options.layout) {
+      const styles = window.getComputedStyle(target)
+      const layoutOffset = styles.getPropertyValue('--v-layout-top')
+
+      if (layoutOffset) targetLocation -= parseInt(layoutOffset, 10)
+    }
+  }
+
+  targetLocation += options.offset
+
+  const startLocation = (horizontal ? container.scrollLeft : container.scrollTop) ?? 0
+
+  if (targetLocation === startLocation) return Promise.resolve(targetLocation)
+
+  const startTime = performance.now()
+
+  return new Promise(resolve => requestAnimationFrame(function step (currentTime: number) {
+    const timeElapsed = currentTime - startTime
+    const progress = Math.abs(options.duration ? Math.min(timeElapsed / options.duration, 1) : 1)
+    const location = Math.floor(startLocation + (targetLocation - startLocation) * ease(progress))
+
+    container[horizontal ? 'scrollLeft' : 'scrollTop'] = location
+
+    if (progress === 1) return resolve(targetLocation)
+
+    let clientSize
+    let reachEnd
+
+    if (!horizontal) {
+      clientSize = container === document.body ? document.documentElement.clientHeight : container.clientHeight
+      reachEnd = clientSize + container.scrollTop >= container.scrollHeight
+
+      if (targetLocation > container.scrollTop && reachEnd) return resolve(targetLocation)
+    } else {
+      clientSize = container === document.body ? document.documentElement.clientWidth : container.clientWidth
+      reachEnd = clientSize + container.scrollLeft >= container.scrollWidth
+
+      if (targetLocation > container.scrollLeft && reachEnd) return resolve(targetLocation)
+    }
+
+    requestAnimationFrame(step)
+  }))
+}
+
+export function useGoTo (_options: Partial<GoToOptions> = {}) {
```

Comment:
Why is this a composable instead of just `import { goTo } from 'vuetify'`?

Reply:
Future flexibility